In [3]:
print("Hello Jesus Vega")

Hello Jesus Vega


Circular Orbits:

a) Starting with Newton's law of Gravitation

$F_g = G\frac{Mm}{r^2}$ where $G = 6.67 x 10^{-11} \frac{m^3}{kgs^2}$ and the mass of the Earth is $M = 5.97 x 10^{24} kg$(1)

the centripetal force is the following

$F_c = \frac{mv^2}{r} = m\omega^2r$ (2)

equate equations (1) and (2) to solve for r

$G\frac{Mm}{r^2} = m\omega^2r$

$r = (\frac{GM}{\omega^2})^\frac{1}{3}$ where $\omega = \frac{2\pi}{T}$ and T is the period

$r = (\frac{GMT^2}{4\pi^2})^\frac{1}{3}$

if we know that the distance to some point is r and the radius of the Earth is R then the height at some point h is the following

$h = r - R$ where $R = 6371 km$

$h = (\frac{GMT^2}{4\pi^2})^\frac{1}{3} - R$

In [23]:
#b) In a Python code box, write a program that prompts the user to enter a value of T in minutes.
    #Verify that the code works with the period of the International Space Stations
    #which has an altitude of 410 km and period of 92.68 minutes.
    
import numpy as np #allows us to call any function from numpy when we call it.
    
G = float(6.67e-11) #units of m^3/kg/s^2
M = float(5.97e24) #units of kg
R_1 = 6371*1000 #converting km to meters
#after python converts the value of R_1 it is called a new variable
R_2 = R_1
#allowing the user to input a value for the period
T_1 = float(input("Enter a value of T in minutes:"))
T_2 = T_1*60
h = ((G*M*(T_2)**2)/(4*np.pi**2))**(1/3) - R_2
    
print(h)

Enter a value of T in minutes:92.68
410696.1325276494


In [25]:
#c) In a new code box, re-code the calculation so it makes the calculation using a user defined function instead.
#Then calculate the altitude of Geosynchronous Orbit where it's one day T = 24 hours.

import numpy as np

G = float(6.67e-11) #units of m^3/kg/s^2
M = float(5.97e24) #units of kg
R_1 = 6371*1000 #converting km to meters
R_2 = R_1
T_1 = 24*3600
T_2 = T_1 #converting hours to seconds
h = ((G*M*(T_2)**2)/((4*np.pi**2)))**(1/3) - R_2

#here we used the define function to redefine the equation for h, we then call the function by returning it
def fun():
    print(h)
fun()

35855910.176174976


In [54]:
#d) In a new code box, write a while loop which repetitively calculates h from T = 90 minutes or 5,400 seconds
#to T = 80 minutes or 4,800 seconds, 
#iterating in 1 minute intervals/ 60 second intervals. 
#Is there a minimal orbital period around Earth

import numpy as np
G = float(6.67e-11) #units of m^3/kg/s^2
M = float(5.97e24) #units of kg
R_1 = 6371*1000 #converting km to meters
R_2 = R_1
T_1 = 4800
T_2 = 5400

#creating a while for values less than or equal to T_2
while (T_1 <= T_2):
    h = ((G*M*(T_1)**2)/((4*np.pi**2)))**(1/3) - R_2 #the equation is under the while loop   
    #reiterates the loop until the value of T_1 equal T_2 with increments of 60 seconds
    T_1 = T_1+60
    #printing the function multiple times while under the while loop
    print("h =", int(h),"meters")

h = -222899 meters
h = -171771 meters
h = -120853 meters
h = -70141 meters
h = -19633 meters
h = 30674 meters
h = 80785 meters
h = 130703 meters
h = 180429 meters
h = 229968 meters
h = 279321 meters


The minimal orbital Period is when the height is the smallest value which is at 19633 meters, which happens at 85 minutes.

Coordinate Conversion:

In [82]:
#a) Write a program which takes x and y as an input and prints out the r and phi (polar angle) of the coordinate. 
#Make sure to utilize if statements to make sure your results are in the correct quadrants.

import numpy as np

#asking the user to input values for x and y
x1 = float(input("Enter a value for x:"))
y1 = float(input("Enter a value for y:"))

#defines what r coordinate is in terms of x and y
r = ((x1**2) + (y1**2))**(1/2)
#equation to find the angle in terms of x and y
theta = np.arctan(y1/x1)

#setting the conditions for when the user puts in negative values for x and y
if x1 < 0 and y1 >0:
        print("r =",float(r),"phi =",f"{theta: .4f}")
elif x1 < 0 and y1 < 0:
        print("r =",float(r),"phi =",-f"{theta: .4f}")
else:
        print("r =",float(r),"phi =",f"{theta: .4f}")

Enter a value for x:3
Enter a value for y:4
r = 5.0 phi =  0.9273


In [76]:
# b) Write a program which takes r and phi and prints out the x and y coordinate. 
#Make sure to utilize if statements to make sure your results are in the correct quadrants.

import numpy as np

r1 = float(input("Enter a value for r:"))
d = float(input("Enter a integer value for phi:"))

#changing the input angle of a float value into radians
phi = d*np.pi/180
#does the calculation for each variable
x1 = r1*np.cos(phi)
y1 = r1*np.sin(phi)

#setting the conditions for each quadrant
#first quadrant should return both positive vales
if   ((phi >= 0))  and ((phi <= np.pi/2)) and ((r >0)):
    print("x =",f"{x1: .4f}", "y =", f"{y1: .4f}")
#second quadrant returns -x but only if r >0 another condition
#has to be written for it as well
elif ((phi >= np.pi/2))  and ((phi <= np.pi)) and ((r < 0)):
    print("x =",f"{x1: .4f}", "y =", f"{-y1: .4f}")
elif ((phi >= np.pi/2))  and ((phi <= np.pi)) and ((r > 0)):
    print("x =",f"{-x1: .4f}", "y =", f"{y1: .4f}")
elif ((phi >= np.pi))  and ((phi <= np.pi*3/2)) and ((r > 0)):
    print("x =",f"{-x1: .4f}", "y =", f"{-y1: .4f}")
elif ((phi >= np.pi))  and ((phi <= np.pi*3/2)) and ((r < 0)):
    print("x =",f"{x1: .4f}", "y =", f"{y1: .4f}")
elif ((phi >= np.pi*3/2))  and ((phi < 0)) and ((r < 0)):
    print("x =",f"{-x1: .4f}", "y =", f"{y1: .4f}")
elif ((phi >= np.pi*3/2))  and ((phi < 0)) and ((r > 0)):
    print("x =",f"{x1: .4f}", "y =", f"{-y1: .4f}")
else:
    print("x =",f"{x1: .4f}", "y =", f"{y1: .4f}")

Enter a value for r:-1
Enter a value for phi:-90
x = -0.0000 y =  1.0000


Lorentz Invariants: The momentum 4-vector and the invariants are:

$p = [\frac{E}{c}, p_x, p_y, p_z]$ (1)

$s = (\frac{E}{c})^2 - p^2_x - p^2_y - p^2_z$ (2)

show that $\sqrt{s} = mc$; where the m is the mass of the particle and c is the speed of light

starting with the energy momentum relation

$(\frac{E}{c^2})^2 = \textbf{p}\cdot\textbf{p} + m^2c^2$

using equation (2) and plugging it into the energy momentum relation we get

$s + p^2_x + p^2_y + p^2_z = \textbf{p}\cdot\textbf{p} + m^2c^2$ 

we can quickly see that $\textbf{p}\cdot\textbf{p} = p^2_x + p^2_y + p^2_z$

we are left with the following relation $s = m^2c^2$

$\sqrt{s} = mc$

In [54]:
#b) Write a function which takes a list [E, px, py, pz] as an input and print out the mass*(speed of light) of the particle.
#Make sure the function returns an obviously wrong answer like -1 for invalid inputs. 
#Extra note, typically in particle physics this energy, momentum and mass all have units of Electrovolt (eV). 
#Mass a proton is 0.938 GeV for example.
# creating an empty list
lst = []
 
# number of elements as input
n = int(input("Enter number of elements : "))
 
# iterating the range of list until the number of elements has been acquired
for i in range(0, n):
    ele = int(input())
    # adding the element
    lst.append(ele) 

 
print(lst)

Enter number of elements : 4
1
2
3
4
[1, 2, 3, 4]


Binding Energy of Isotopes:

In [101]:
#listing the values for each variable
import numpy as np

a1 = 15.8
a2 = 18.3
a3 = 0.714
a4 = 23.2

A = 58#float(int("Enter a value for A:"))
Z = 28#float(int("Enter a value for Z:"))

n == int

if A == 2*n +1:
    a5 = 0
    E_b = a1*A - a2*A**2/3 - a3*Z**2/A**1/3 - a4*(A-2*z)**2/A + a5/A**1/2
    print(E_b)
elif A and Z == 2*n:
    a5 = 12.0
    E_b = a1*A - a2*A**2/3 - a3*Z**2/A**1/3 - a4*(A-2*z)**2/A + a5/A**1/2
    print(E_b)
elif A == 2*n and Z == 2*n +1:
    E_b = a1*A - a2*A**2/3 - a3*Z**2/A**1/3 - a4*(A-2*z)**2/A + a5/A**1/2
    print(E_b)
    